callbacks.py

```diff
@@ -105,6 +105,23 @@ def show_raw_data(ventilator_data, delete):
 
     return children_vent, hidden, filename
 
+@callback(Output('overlaid', 'children'),
+          Input('hidden-div', 'data'),)
+def show_raw_data(emg_data, delete):
+    emg_data = variables.get_emg()
+    hidden = True
+
+    trigger_id = ctx.triggered_id
+    filename = variables.get_emg_filename()
+
+    if emg_data is not None:
+        emg_frequency = variables.get_emg_freq()
+        children_emg = utils.add_emg_graphs(np.array(emg_data), emg_frequency)
+        hidden = False
+    else:
+        children_emg = []
+
+    return children_emg, hidden, filename

```

Comment:
The arguments  in the `show_raw_data` function are the `Input` and `State` provided in the decorator, while the values returned are assigned to the `Output`. So, the number of arguments has to match the number of inputs and the number of returned values has to match the number of outputs.

```suggestion
@callback(Output('overlaid', 'children'),
          Input('hidden-div', 'data'),)
def show_raw_data(input):
    emg_data = variables.get_emg()

    if emg_data is not None:
        emg_frequency = variables.get_emg_freq()
        children_emg = utils.add_emg_graphs(np.array(emg_data), emg_frequency)
    else:
        children_emg = []

    return children_emg

```

Reply:
In general, I think it would be better to have a callbacks script per each page. It was not the case for the `upload_data` and the `view_raw` pages, but I fixed it in [this](https://github.com/ReSurfEMG/ReSurfEMG-dashboard/pull/14) PR.

Reply:
OK, I'll try writing it with callback script for the page, then putting in a PR for that,but fundamentally this is just a placeholder. The callback was actually not correct